crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -297,102 +305,212 @@ impl TransactionBlock {
+                        },
+                        FqNameFilter::ByFqName(p, m, n) => (
+                            select_fun(p, m.clone(), n.clone(), sender, &bound),
+                            "tx_calls_fun",
+                        ),
+                    });
+                }
+                if let Some(kind) = &filter.kind {
+                    subqueries.push((select_kind(*kind, &bound), "tx_kinds"));
+                    if let Some(sender) = &sender {
+                        subqueries.push((select_sender(sender, &bound), "tx_senders"));
+                    }
+                }
+                if let Some(recv) = &filter.recv_address {
+                    subqueries.push((select_recipient(recv, sender, &bound), "tx_recipients"));
+                }
+                if let Some(input) = &filter.input_object {
+                    subqueries.push((select_input(input, sender, &bound), "tx_input_objects"));
+                }
+                if let Some(changed) = &filter.changed_object {
+                    subqueries.push((
+                        select_changed(changed, sender, &bound),
+                        "tx_changed_objects",
+                    ));
+                }
+
+                if filter.num_filters() == 0 {
+                    // We don't need to explicitly specify sender unless it's the only filter
+                    if let Some(sender) = &sender {
+                        subqueries.push((select_sender(sender, &bound), "tx_senders"));
+                    }
+                    // And if there are no filters at all, we can operate directly on the main table
+                    else {
+                        subqueries.push((
+                            select_tx(None, &bound, "transactions"),
+                            "transactions",
+                        ));
+                    }
+                }
+
+                // Finally, join the array of subqueries into a single query that looks like: SELECT
+                // tx_sequence_number FROM (SELECT tx_sequence_number FROM table0 WHERE ...) AS
+                // table0 INNER JOIN (SELECT ...) AS table1 USING (tx_sequence_number) ORDER BY
+                // tx_sequence_number ASC LIMIT 52;
+
+                // There should be at least one subquery even if no filters are specified - the
+                // query against the base table. Otherwise the array will contain subqueries that
+                // hit lookup tables.
+                let mut subquery = subqueries.pop().unwrap().0;
+
+                if !subqueries.is_empty() {
+                    subquery = query!(
+                        "SELECT tx_sequence_number FROM ({}) AS initial",
+                        subquery
+                    );
+                    while let Some((subselect, alias)) = subqueries.pop() {
+                        subquery =
+                            query!("{} INNER JOIN {} USING (tx_sequence_number)", subquery, aliased => (subselect, alias));
```

Comment:
Going to my earlier comment -- I think the more general concept you want is an `inner_join!` macro that accepts two variants (with and without an alias), then you would write this as:

```rust
subquery = inner_join!(subquery, alias => subselect, using: ["tx_sequence_number"])
```

Reply:
ah dope yes, thanks for the suggestion!